pkg/autoupgrade/daemon_test.go

```diff
@@ -337,6 +341,22 @@ func TestRefreshImages(t *testing.T) {
 			appKeysPrevCheckAfter:  map[kclient.ObjectKey]time.Time{router.Key("acorn", "notify-app"): now},
 			appsUpdated:            map[string]string{"notify-app": "docker.io/acorn/notify"},
 		},
+		{
+			name:                   "Auto refresh tag with multiple remote tags with latest tag image denied and current being an older image",
+			client:                 &mockDaemonClient{remoteTags: []string{"v1.1.1", "v1.1.2"}, imageDenyList: map[string]any{"v1.1.2": true}},
+			appKeysPrevCheckBefore: map[kclient.ObjectKey]time.Time{router.Key("acorn", "test-1"): thirtySecondsAgo},
+			imagesToRefresh:        map[imageAndNamespaceKey][]kclient.ObjectKey{{image: "index.docker.io/acorn/test-1:v1.1.1", namespace: "acorn"}: {router.Key("acorn", "test-1")}},
+			appKeysPrevCheckAfter:  map[kclient.ObjectKey]time.Time{router.Key("acorn", "test-1"): now},
+			appsUpdated:            map[string]string{"test-1": "index.docker.io/acorn/test-1:v1.1.2"},
+		},
```

Comment:
I would expect that if the current tag is `v1.1.1` and the tag `v1.1.2` is denied, then the app would not have its image updated. The issue, I believe, is that the `imageDenyList` keys should be the full image: `index.docker.io/acorn/test-1:v1.1.2`.

Note that doing this in the test below reveals a bug in your implementation.

Reply:
Got it, yeah i see that now. Just wanted to clarify that by reveals a bug in your implementation you mean the testing code logic correct?

Reply:
I realized that not this testcase but in the next one once i fix the image name in the imageDenyList, the `findLatestTagForImageWithPattern` keeps looping currently fixing that logic 👍